flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReader.java

```diff
@@ -414,6 +433,40 @@ private void maybeRegisterKafkaConsumerMetrics(
         }
     }
 
+    private Supplier<Long> getBytesConsumedTotalSupplier(KafkaConsumer<?, ?> consumer) {
```

Comment:
Nit: You can directly return the `Optional<Supplier<Long>>` to replace the null return.

Reply:
The compiler complains a warn if I use ```Optional``` as a class field 🤔 I'd like to keep it for now (forgive me being neat-freak lol)